src/sync_protocol/src/agent_sync_protocol.cpp

```diff
@@ -444,3 +502,29 @@ void AgentSyncProtocol::clearSyncState()
     std::lock_guard<std::mutex> lock(m_syncState.mtx);
     m_syncState.reset();
 }
+
+std::vector<PersistedData> AgentSyncProtocol::filterDataByRanges(
+    const std::vector<PersistedData>& sourceData,
+    const std::vector<std::pair<uint64_t, uint64_t>>& ranges)
+{
+    std::vector<PersistedData> result;
+
+    if (ranges.empty() || sourceData.empty())
+    {
+        return result;
+    }
+
+    for (const auto& item : sourceData)
+    {
+        for (const auto& range : ranges)
+        {
+            if (item.seq >= range.first && item.seq <= range.second)
```

Comment:
Maybe we can also validate that the second element of the pair is greater or equal than the first one?